Some Portlet names breaks the app

I've defined `VaadinIPCPortlet` class.
It broke the app with the exception:

```
20-Nov-2019 10:32:19.219 SEVERE [localhost-startStop-1] org.apache.catalina.core.StandardContext.startInternal Error during ServletContainerInitializer processing
 javax.servlet.ServletException: WebComponentConfigurationRegistryInitializer failed to collect WebComponentExporter implementations!
	at com.vaadin.flow.server.startup.WebComponentConfigurationRegistryInitializer.onStartup(WebComponentConfigurationRegistryInitializer.java:80)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5225)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:754)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:730)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:734)
	at org.apache.catalina.startup.HostConfig.deployWAR(HostConfig.java:980)
	at org.apache.catalina.startup.HostConfig$DeployWar.run(HostConfig.java:1851)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.vaadin.flow.server.InvalidCustomElementNameException: Tag name 'vaadin-iPC-portlet' given by 'com.vaadin.flow.portal.events.VaadinIPCPortlet' is not a valid custom element name.
	at com.vaadin.flow.server.startup.WebComponentConfigurationRegistryInitializer.validateTagNames(WebComponentConfigurationRegistryInitializer.java:112)
	at com.vaadin.flow.server.startup.WebComponentConfigurationRegistryInitializer.onStartup(WebComponentConfigurationRegistryInitializer.java:75)
	... 12 more
```

The logic which derive the portlet tag name should be fixed.

--- a/vaadin-portlet/src/main/java/com/vaadin/flow/portal/VaadinPortlet.java
+++ b/vaadin-portlet/src/main/java/com/vaadin/flow/portal/VaadinPortlet.java
@@ -433,6 +433,7 @@ public abstract class VaadinPortlet<C extends Component> extends GenericPortlet
             }
             candidate = candidate.replace(".-", "-");
             candidate = candidate.replace('.', '-');
+            candidate = candidate.replace('$', '-');
             return candidate;
         }
     }

--- a/vaadin-portlet/src/test/java/com/vaadin/flow/portal/VaadinPortletTest.java
+++ b/vaadin-portlet/src/test/java/com/vaadin/flow/portal/VaadinPortletTest.java
@@ -74,6 +74,10 @@ public class VaadinPortletTest {
 
     }
 
+    private static class Special$Character extends VaadinPortlet<Div> {
+
+    }
+
     private String namespace = "namespace-foo";
 
     private TestComponent component;
@@ -324,5 +328,13 @@ public class VaadinPortletTest {
         TestMYPortlet portlet = new TestMYPortlet();
         String tag = portlet.getTag();
         Assert.assertNotEquals(tag, new Wrapper.TestMYPortlet().getTag());
+    }
+
+    @Test
+    public void getTag_tagNameDoNoContainUpperCaseLettersAndDollarSign() {
+        Special$Character portlet = new Special$Character();
+        String tag = portlet.getTag();
+        Assert.assertFalse(tag.chars().anyMatch(Character::isUpperCase));
+        Assert.assertFalse(tag.chars().anyMatch(ch -> ch == '$'));
     }
 }

